### PR TITLE
Story3

### PR DESCRIPTION
**Pull request by FinHardy**
Added functionality for working text box to remove car by typing in the name and submitting the form

### CODE REVIEW

File: src/CarModel.php [hunk 1]
```diff
@@ -45,4 +45,24 @@ public function addCar(Car $car): bool
...
+    {
+        if (!empty($this->searchForName($carname))) {
+            $query = $this->db->prepare("UPDATE `cars` SET `deleted` = 1 WHERE `name` = :name ;");
+            $query->bindParam('name', $carname);
+
+            return $query->execute();
+        } else {
+            return false;
+        }
+    }
```
Comment (ashleycoles) [inline src/CarModel.php]: You should really do deleting by id rather than name, it's much more reliable

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: src/utils.php [hunk 2]
```diff
@@ -36,35 +36,56 @@ function create_list_of_cars(array $cars): string | false
...
+        echo "Name cannot be over 30 characters long" . "<br>";
+        $ret = false;
     }
 
-
-    if (empty($year_made) || !is_numeric($year_made) || strlen(strval($year_made)) <= 0 || strlen(strval($year_made)) > 4 || $year_made < 0 || $year_made > 2023) {
-        $err_msg .= "Invalid year" . "<br>";
+    if (
+        empty($car->year_made) == true ||
+        is_int($car->year_made) == false ||
```
Comment (ashleycoles) [inline src/utils.php]: Some of these undeeded comparisons have crept back in

```
empty($car->year_made) ||
!is_int($car->year_made) ||
```

File: src/utils.php [hunk 3]
```diff
@@ -36,35 +36,56 @@ function create_list_of_cars(array $cars): string | false
...
+        strlen(strval($car->year_made)) <= 0 ||
+        strlen(strval($car->year_made)) > 4 ||
+        $car->year_made < 0 ||
+        $car->year_made > 2023
+    ) {
+        echo "Invalid year" . "<br>";
```
Comment (ashleycoles) [inline src/utils.php]: Dont forget you shouldn't be echoing from within a function, it makes it hard to test

**Review by ashleycoles — CHANGES_REQUESTED** [review]
So close now!

File: src/utils.php (hunk 2 above)
Comment (FinHardy) [reply src/utils.php]: This is weird... I made this change and pulled from the branch with these changes... git being weird or me being a noob probably. anyways ill fix it now

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**